bbb-common-message/src/main/protobuf/meeting-service.proto

```diff
@@ -0,0 +1,96 @@
+syntax = "proto3";
+
+package org.bigbluebutton.protos;
+
+message MeetingRunningReq {
+  string meeting_id = 1;
+}
+
+message MeetingRunningResp {
+  bool is_running = 1;
+}
+
+message Attendee {
+  string user_id = 1;
+  string full_name = 2;
+  string role = 3;
+  bool is_presenter = 4;
+  bool is_listening_only = 5;
+  bool has_joined_voice = 6;
+  bool has_video = 7;
+  string clientType = 8;
+  map<string, string> custom_data = 9;
+}
+
+message DurationInfo {
+  int64 create_time = 1;
+  string created_on = 2;
+  int32 duration = 3;
+  int64 start_time = 4;
+  int64 end_time = 5;
+  bool is_running = 6;
+  bool has_been_forcibly_ended = 7;
+}
+
+message ParticipantInfo {
+  bool has_user_joined = 1;
+  int32 participant_count = 2;
+  int32 listener_count = 3;
+  int32 voice_participant_count = 4;
+  int32 video_count = 5;
+  int32 max_users = 6;
+  int32 moderator_count = 7;
+}
+
+message BreakoutInfo {
+  bool is_breakout = 1;
+  string parent_meeting_id = 2;
+  int32 sequence = 3;
+  bool free_join = 4;
+}
+
+message MeetingInfo {
+  string meeting_name = 1;
+  string meeting_ext_id = 2;
+  string meeting_int_id = 3;
+  string voice_bridge = 6;
+  string dial_number = 7;
+  string attendee_pw = 8;
+  string moderator_pw = 9;
+  bool recording = 10;
+  repeated Attendee attendees = 11;
+  map<string, string> metadata = 12;
+  repeated string breakout_rooms = 13;
+  DurationInfo duration_info = 14;
+  ParticipantInfo participant_info = 15;
+  BreakoutInfo breakout_info = 16;
+}
+
+message MeetingInfoReq {
+  string meeting_id = 1;
+}
+
+message MeetingInfoResp {
+  MeetingInfo  meeting_info = 1;
+}
+
+message ListMeetingsReq {
+  int32 page_size = 1;
+  string page_token = 2;
+}
+
+message ListMeetingsResp {
+  repeated MeetingInfo meetings = 1;
+  string next_page_token = 2;
+}
+
+message StreamMeetingsReq {
+  string meeting_id = 1;
+}
+
+service MeetingService {
+  rpc isMeetingRunning(MeetingRunningReq) returns (MeetingRunningResp);
+  rpc getMeetingInfo(MeetingInfoReq) returns (MeetingInfoResp);
+  rpc listMeetings(ListMeetingsReq) returns (ListMeetingsResp);
+  rpc streamMeetings(StreamMeetingsReq) returns (stream MeetingInfoResp);
```

Comment:
The name `stream` gives the impression that the channel remains open to receive the meeting content in real-time, which is not actually how it functions. Could we consider renaming it to prevent this misunderstanding? Alternatively, if there's a specific reason behind choosing this name,  knowing it might help me better understand the choice.

Reply:
The reason for the name 'streamMeetings' is that this request uses server-side streaming to get all of the running meetings from `akka-apps`. A single request is sent from `bbb-web` and then a stream of `MeetingInfoResp` corresponding to each meeting is returned sequentially.

Reply:
I suggest renaming the message to `getMeetings` as it more precisely conveys the intention behind the message. Including "stream" in the name seems to be a technical specification, which could perhaps be better suited for a detailed comment, rather than the message name itself.

Reply:
I see. My issue with the name `getMeetings` is that it is a name that would typically be used for a unary call where the client sends a single request and then the server bundles all of the meetings together into a single response and then returns them. While we don't have such a function right now, and it is generally not advisable to return such large responses, in the future it might be necessary to add such a function and I would like to reserve the name `getMeetings` for such a case.

Another thing is that I feel the name `streamMeetings` is actually much more precise in conveying the purpose of the request and allows someone implementing a gRPC client that will make that request to understand that they will be in fact opening a stream between the client and the server and getting back a series of responses rather than a single response. For example, this will allow them to know that ideally such a request should be handled in an asynchronous manner on the client side and that a `StreamObserver` will be required to manage the responses.

An alternative name to `streamMeetings` could be something like `getMeetingStream` if that is preferable since it would still convey the fact that the client is not getting back a single response from the server.

Reply:
Alright!
I like the name `getMeetingStream`. Maybe `getMeetingsStream` in the plural?